README.md
## Currency API PHP Library

[![Build Status](https://github.com/horisystems/fxdatapi-php/actions/workflows/ci.yml/badge.svg?branch=main)](https://github.com/horisystems/fxdatapi-php/actions?query=branch%3Amain)
[![Latest Stable Version](http://poser.pugx.org/horisystems/fxdatapi/v)](https://packagist.org/packages/horisystems/fxdatapi)
[![Total Downloads](https://poser.pugx.org/horisystems/fxdatapi/downloads.svg)](https://packagist.org/packages/horisystems/fxdatapi)
[![License](https://poser.pugx.org/horisystems/fxdatapi/license.svg)](https://packagist.org/packages/horisystems/fxdatapi)

The SDK provides convenient access to the [Currency API](https://horisystems.com/currency-api/) for applications written in the [PHP](https://www.php.net/) Programming Language.

### Requirements

PHP 7.x.x and later.

### Dependencies

The library require the following extensions in order to work properly:

-   [`Guzzle`](https://docs.guzzlephp.org/en/stable/)

> If you use Composer, these dependencies should be handled automatically. If you install manually, you'll want to make sure that these extensions are available.

### Development

Instructions for installing Composer on macOS, Linux, Unix and Windows are available [here](https://getcomposer.org/doc/00-intro.md#installation-windows).

### Install dependencies:

```bash
composer install
```

### Installation

Install Currency API PHP SDK by running the following command:

```sh
composer require horisystems/fxdatapi
```

### Usage Example

```php
## create a file called currency_api.php
<?php
use Fxdatapi\Fxdatapi;

// Initialize the Fxdatapi SDK
$fxdatapi = new Fxdatapi();

// Set your username and password for the Currency API
$username = 'your_username';
$password = 'your_password';

// Log in to the API and obtain a token (if necessary)
$loginResult = $fxdatapi->login($username, $password);
if ($loginResult) {
    echo "Login successful.\n";
} else {
    die("Login failed.\n");
}

// Set currency conversion details
$date = '2023_04_05';
$baseCurrency = 'GBP';
$targetCurrency = 'EUR';
$amount = 1000;

// Set the date for which you want to retrieve the currencies
$day = '05';
$month = '04';
$year = '2023';

// Set the UUID of the currency you want to retrieve
$uuid = 'currency_uuid';

// Set the date for which you want to retrieve the daily average
$dailyAverageDate = '2023_04_10';

// Set the date range for which you want to retrieve the weekly average
$weeklyAverageFromDate = '2023_04_03';
$weeklyAverageToDate = '2023_04_07';

// Set the UUID of the performance you want to retrieve
$performanceId = 'performance_id';

// Set the UUID of the signal you want to retrieve
$signalId = 'signal_id';

// Call the convert method to perform the currency conversion
$conversionResult = $fxdatapi->convert($username, $date, $baseCurrency, $targetCurrency, $amount);
if ($conversionResult) {
    echo "Conversion result: " . json_encode($conversionResult) . "\n";
} else {
    echo "Currency conversion failed: " . $conversionResult['error'] . "\n";
}

// Call the convertAll method to convert the base currency to all available target currencies
$convertAllResult = $fxdatapi->convertAll($username, $baseCurrency, $amount, $date);
if ($convertAllResult) {
    echo "Conversion to all currencies successful:\n";
    echo json_encode($convertAllResult) . "\n";
} else {
    echo "Conversion to all currencies failed: " . $convertAllResult['error'] . "\n";
}

// Call the getCurrencies method to get the available currencies for the given date
$currenciesResult = $fxdatapi->getCurrencies($username, $day, $month, $year);
if ($currenciesResult) {
    echo "Currencies for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($currenciesResult) . "\n";
} else {
    echo "Failed to retrieve currencies for the given date.\n";
}

// Call the getCurrency method to get the details of the specific currency for the given date
$currencyResult = $fxdatapi->getCurrency($uuid, $username, $day, $month, $year);
if ($currencyResult) {
    echo "Currency details for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($currencyResult) . "\n";
} else {
    echo "Failed to retrieve currency details for the given date.\n";
}

// Call the getHistoricalData method to get historical data for all currencies for the given date
$historicalDataResult = $fxdatapi->getHistoricalData($username, $date, $day, $month, $year);
if ($historicalDataResult) {
    echo "Historical data for " . $date . ":\n";
    echo json_encode($historicalDataResult) . "\n";
} else {
    echo "Failed to retrieve historical data for the given date.\n";
}

// Call the getHistoricalDataForCurrency method to get historical data for a specific currency
$historicalDataForCurrencyResult = $fxdatapi->getHistoricalDataForCurrency($uuid, $username, $day, $month, $year, $date);
if ($historicalDataForCurrencyResult) {
    echo "Historical data for currency " . $uuid . " on " . $date . ":\n";
    echo json_encode($historicalDataForCurrencyResult) . "\n";
} else {
    echo "Failed to retrieve historical data for the given currency and date.\n";
}

// Call the getDailyAverage method to get the daily average for the given date
$dailyAverageResult = $fxdatapi->getDailyAverage($dailyAverageDate);
if ($dailyAverageResult) {
    echo "Daily average for " . $dailyAverageDate . ":\n";
    echo json_encode($dailyAverageResult) . "\n";
} else {
    echo "Failed to retrieve daily average for the given date.\n";
}

// Call the getWeeklyAverage method to get the weekly average for the given date range
$weeklyAverageResult = $fxdatapi->getWeeklyAverage($weeklyAverageFromDate, $weeklyAverageToDate);
if ($weeklyAverageResult) {
    echo "Weekly average from " . $weeklyAverageFromDate . " to " . $weeklyAverageToDate . ":\n";
    echo json_encode($weeklyAverageResult) . "\n";
} else {
    echo "Failed to retrieve weekly average for the given date range.\n";
}

// Call the getMonthlyAverage method to get the monthly average for the given year and month
$monthlyAverageResult = $fxdatapi->getMonthlyAverage($monthlyAverageYear, $monthlyAverageMonth);
if ($monthlyAverageResult) {
    echo "Monthly average for " . $monthlyAverageYear . " and " . $monthlyAverageMonth . ":\n";
    echo json_encode($monthlyAverageResult) . "\n";
} else {
    echo "Failed to retrieve monthly average for the given year and month.\n";
}

// Call the getMarginsSpreads method to get the margins and spreads for the given date
$marginsSpreadsResult = $fxdatapi->getMarginsSpreads($username, $day, $month, $year);
if ($marginsSpreadsResult) {
    echo "Margins and spreads for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($marginsSpreadsResult) . "\n";
} else {
    echo "Failed to retrieve margins and spreads for the given date.\n";
}

// Call the getMarginSpread method to get the details of the specific margin or spread for the given date
$marginSpreadResult = $fxdatapi->getMarginSpread($uuid, $username, $day, $month, $year);
if ($marginSpreadResult) {
    echo "Margin or spread details for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($marginSpreadResult) . "\n";
} else {
    echo "Failed to retrieve margin or spread details for the given date.\n";
}

// Call the getAllPerformances method to get all performances
$getAllPerformancesResult = $fxdatapi->getAllPerformances($username);
if ($getAllPerformancesResult) {
    echo "Performances list:\n";
    echo json_encode($getAllPerformancesResult) . "\n";
} else {
    echo "Failed to retrieve performances.\n";
}

// Call the getPerformanceById method to get the details of a specific performance
$getPerformanceByIdResult = $fxdatapi->getPerformanceById($performanceId, $username);
if ($getPerformanceByIdResult) {
    echo "Performance details for " . $performanceId . ":\n";
    echo json_encode($getPerformanceByIdResult) . "\n";
} else {
    echo "Failed to retrieve performance details for the given ID.\n";
}

// Call the getAllSignals method to get all signals
$getAllSignalsResult = $fxdatapi->getAllSignals($username);
if ($getAllSignalsResult) {
    echo "Signals list:\n";
    echo json_encode($getAllSignalsResult) . "\n";
} else {
    echo "Failed to retrieve signals.\n";
}

// Call the getSignalById method to get the details of a specific signal
$getSignalByIdResult = $fxdatapi->getSignalById($signalId, $username);
if ($getSignalByIdResult) {
    echo "Signal details for " . $signalId . ":\n";
    echo json_encode($getSignalByIdResult) . "\n";
} else {
    echo "Failed to retrieve signal details for the given ID.\n";
}

?>
```

Execute the `currency_api.php` file:

```sh
php currency_api.php
```

### Setting up Currency API Account

Subscribe [here](https://horisystems.com/currency-api/) for a user account.

### Using the Currency API

You can read the [API documentation](https://docs.fxdatapi.com/) to understand what’s possible with the Currency API. If you need further assistance, don’t hesitate to [contact us](https://horisystems.com/contact/).

### License

This project is licensed under the [BSD 3-Clause License](./LICENSE).

### Copyright

(c) 2023 [Hori Systems Limited](https://horisystems.com).

CHANGELOG.md
# 1.0.9 (October 28, 2023)

* Business Changes.

# 1.0.8 (May 10, 2023)

* Monthly Average added.

# 1.0.7 (May 03, 2023)

* Monthly Average added.

# 1.0.6 (April 27, 2023)

* Performances and Signals functions added.

tests/FxdatapiTest.php
<?php

namespace Fxdatapi\Tests;

use PHPUnit\Framework\TestCase;
use Fxdatapi\Fxdatapi;

class FxdatapiTest extends TestCase
{
    protected $fxdatapi;

    protected function setUp(): void
    {
        $this->fxdatapi = new Fxdatapi();
    }

    public function testLogin()
    {
        $username = 'your_username';
        $password = 'your_password';

        $result = $this->fxdatapi->login($username, $password);
        $this->assertTrue(is_array($result));
        $this->assertArrayHasKey('message', $result);
    }

    public function testConvert()
    {
        $username = 'your_username';
        $date = '2023_04_05';
        $baseCurrency = 'GBP';
        $targetCurrency = 'EUR';
        $amount = 1000;

        $result = $this->fxdatapi->convert($username, $date, $baseCurrency, $targetCurrency, $amount);
        $this->assertTrue(is_array($result));
    }

    public function testConvertAll()
    {
        $username = 'your_username';
        $baseCurrency = 'GBP';
        $amount = 1000;
        $date = '2023_04_05';

        $result = $this->fxdatapi->convertAll($username, $baseCurrency, $amount, $date);
        $this->assertTrue(is_array($result));
    }

    public function testGetCurrencies()
    {
        $username = 'your_username';
        $day = '05';
        $month = '04';
        $year = '2023';

        $result = $this->fxdatapi->getCurrencies($username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetCurrency()
    {
        $uuid = 'currency_uuid';
        $username = 'your_username';
        $day = '05';
        $month = '04';
        $year = '2023';

        $result = $this->fxdatapi->getCurrency($uuid, $username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetHistoricalData()
    {
        $username = 'your_username';
        $date = '2023_04_05';
        $day = '05';
        $month = '04';
        $year = '2023';

        $result = $this->fxdatapi->getHistoricalData($username, $date, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetHistoricalDataForCurrency()
    {
        $uuid = 'currency_uuid';
        $username = 'your_username';
        $day = '05';
        $month = '04';
        $year = '2023';
        $date = '2023_04_05';

        $result = $this->fxdatapi->getHistoricalDataForCurrency($uuid, $username, $day, $month, $year, $date);
        $this->assertTrue(is_array($result));
    }

    public function testGetDailyAverage()
    {
        $date = '2023_04_05';
    
        $result = $this->fxdatapi->getDailyAverage($date);
        $this->assertTrue(is_array($result));
    }
    
    public function testGetWeeklyAverage()
    {
        $from_date = '2023_04_03';
        $to_date = '2023_04_07';
    
        $result = $this->fxdatapi->getWeeklyAverage($from_date, $to_date);
        $this->assertTrue(is_array($result));
    }

    public function testGetMonthlyAverage()
    {
        $year = '2023';
        $month = '04';
    
        $result = $this->fxdatapi->getMonthlyAverage($year, $month);
        $this->assertTrue(is_array($result));
    }

    public function testGetMarginsSpreads()
    {
        $username = 'your_username';
        $day = '19';
        $month = '04';
        $year = '2023';

        $result = $this->fxdatapi->getMarginsSpreads($username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetMarginSpread()
    {
        $uuid = 'uuid_uuid';
        $username = 'your_username';
        $day = '19';
        $month = '04';
        $year = '2023';

        $result = $this->fxdatapi->getMarginSpread($uuid, $username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetAllPerformances()
    {
        $username = 'your_username';

        $result = $this->fxdatapi->getAllPerformances($username);
        $this->assertTrue(is_array($result));
    }

    public function testGetPerformanceById()
    {
        $uuid = 'd4762c44-e3c6-11ed-8570-acde48001122';
        $username = 'your_username';

        $result = $this->fxdatapi->getPerformanceById($uuid, $username);
        $this->assertTrue(is_array($result));
    }

    public function testGetAllSignals()
    {
        $username = 'your_username';

        $result = $this->fxdatapi->getAllSignals($username);
        $this->assertTrue(is_array($result));
    }

    public function testGetSignalById()
    {
        $uuid = 'd46cc05a-e3c6-11ed-8570-acde48001122';
        $username = 'your_username';

        $result = $this->fxdatapi->getSignalById($uuid, $username);
        $this->assertTrue(is_array($result));
    }
}


tests/CurrenseesTest.php
<?php

namespace Currensees\Tests;

use PHPUnit\Framework\TestCase;
use Currensees\Currensees;

class CurrenseesTest extends TestCase
{
    protected $currensees;

    protected function setUp(): void
    {
        $this->currensees = new Currensees();
    }

    public function testLogin()
    {
        $username = 'your_username';
        $password = 'your_password';

        $result = $this->currensees->login($username, $password);
        $this->assertTrue(is_array($result));
        $this->assertArrayHasKey('message', $result);
    }

    public function testConvert()
    {
        $username = 'your_username';
        $date = '2023_04_05';
        $baseCurrency = 'GBP';
        $targetCurrency = 'EUR';
        $amount = 1000;

        $result = $this->currensees->convert($username, $date, $baseCurrency, $targetCurrency, $amount);
        $this->assertTrue(is_array($result));
    }

    public function testConvertAll()
    {
        $username = 'your_username';
        $baseCurrency = 'GBP';
        $amount = 1000;
        $date = '2023_04_05';

        $result = $this->currensees->convertAll($username, $baseCurrency, $amount, $date);
        $this->assertTrue(is_array($result));
    }

    public function testGetCurrencies()
    {
        $username = 'your_username';
        $day = '05';
        $month = '04';
        $year = '2023';

        $result = $this->currensees->getCurrencies($username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetCurrency()
    {
        $uuid = 'currency_uuid';
        $username = 'your_username';
        $day = '05';
        $month = '04';
        $year = '2023';

        $result = $this->currensees->getCurrency($uuid, $username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetHistoricalData()
    {
        $username = 'your_username';
        $date = '2023_04_05';
        $day = '05';
        $month = '04';
        $year = '2023';

        $result = $this->currensees->getHistoricalData($username, $date, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetHistoricalDataForCurrency()
    {
        $uuid = 'currency_uuid';
        $username = 'your_username';
        $day = '05';
        $month = '04';
        $year = '2023';
        $date = '2023_04_05';

        $result = $this->currensees->getHistoricalDataForCurrency($uuid, $username, $day, $month, $year, $date);
        $this->assertTrue(is_array($result));
    }

    public function testGetDailyAverage()
    {
        $date = '2023_04_05';
    
        $result = $this->currensees->getDailyAverage($date);
        $this->assertTrue(is_array($result));
    }
    
    public function testGetWeeklyAverage()
    {
        $from_date = '2023_04_03';
        $to_date = '2023_04_07';
    
        $result = $this->currensees->getWeeklyAverage($from_date, $to_date);
        $this->assertTrue(is_array($result));
    }

    public function testGetMonthlyAverage()
    {
        $year = '2023';
        $month = '04';
    
        $result = $this->currensees->getMonthlyAverage($year, $month);
        $this->assertTrue(is_array($result));
    }

    public function testGetMarginsSpreads()
    {
        $username = 'your_username';
        $day = '19';
        $month = '04';
        $year = '2023';

        $result = $this->currensees->getMarginsSpreads($username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetMarginSpread()
    {
        $uuid = 'uuid_uuid';
        $username = 'your_username';
        $day = '19';
        $month = '04';
        $year = '2023';

        $result = $this->currensees->getMarginSpread($uuid, $username, $day, $month, $year);
        $this->assertTrue(is_array($result));
    }

    public function testGetAllPerformances()
    {
        $username = 'your_username';

        $result = $this->currensees->getAllPerformances($username);
        $this->assertTrue(is_array($result));
    }

    public function testGetPerformanceById()
    {
        $uuid = 'd4762c44-e3c6-11ed-8570-acde48001122';
        $username = 'your_username';

        $result = $this->currensees->getPerformanceById($uuid, $username);
        $this->assertTrue(is_array($result));
    }

    public function testGetAllSignals()
    {
        $username = 'your_username';

        $result = $this->currensees->getAllSignals($username);
        $this->assertTrue(is_array($result));
    }

    public function testGetSignalById()
    {
        $uuid = 'd46cc05a-e3c6-11ed-8570-acde48001122';
        $username = 'your_username';

        $result = $this->currensees->getSignalById($uuid, $username);
        $this->assertTrue(is_array($result));
    }
}


.php-cs-fixer.php
<?php

$finder = PhpCsFixer\Finder::create()
    ->notPath('tests/FxdatapiTest.php');

$config = new PhpCsFixer\Config();
$config->setRiskyAllowed(true);
$config->setRules([
    // Rulesets
    '@PSR2' => true,
    '@PhpCsFixer' => true,
    '@PhpCsFixer:risky' => true,
    '@PHP56Migration:risky' => true,
    '@PHPUnit57Migration:risky' => true,

    // Additional rules
    'fopen_flags' => true,
    'linebreak_after_opening_tag' => true,
    // This one is non-deterministic based on what environment you are running it in and what `get_defined_constants` returns.
    'native_constant_invocation' => false,
    'native_function_invocation' => [
        "strict" => false,
    ],

    // --- Diffs from @PhpCsFixer / @PhpCsFixer:risky ---

    // This is the same as the default for the @PhpCsFixer ruleset, minus
    // the following values: ['include', 'include_once', 'require',
    // 'require_once']. We could enable them and remove this line after
    // updating codegen for the `init.php` file to be compliant.
    'blank_line_before_statement' => ['statements' => ['break', 'case', 'continue', 'declare', 'default', 'exit', 'goto', 'return', 'switch', 'throw', 'try']],

    // This is just prettier / easier to read.
    'concat_space' => ['spacing' => 'one'],

    // This causes strange ordering with codegen'd classes. We might be
    // able to enable this if we update codegen to output class elements
    // in the correct order.
    'ordered_class_elements' => false,

    // Keep this disabled to avoid unnecessary diffs in PHPDoc comments of
    // codegen'd classes.
    'phpdoc_align' => false,

    // This is a "risky" rule that causes a bug in our codebase.
    'self_accessor' => false,

    // Visibility annotations are not supported by php5.6
    'visibility_required' => false,

    // Apparently "uninitialized" is distinct from "null" in some versions of PHP
    // so I am defensively disabling this rule so as to not cause breaking changes
    // but we can feel free to remove it in a major version (or maybe in a minor if
    // we devote some effort into determining that it is safe)
    'no_null_property_initialization' => false,

    // We have to support `@return void` to satisfy Symfony deprecations helper.
    'phpdoc_no_empty_return' => false,
]);
$config->setFinder($finder);
return $config;

example.php
<?php

require_once './src/fxdatapi.php';

use Fxdatapi\Fxdatapi;

// Initialize the Fxdatapi SDK
$fxdatapi = new Fxdatapi();

// Set your username and password for the Currency API
$username = 'your_username';
$password = 'your_password';

// Log in to the API and obtain a token (if necessary)
$loginResult = $fxdatapi->login($username, $password);
if ($loginResult) {
    echo "Login successful.\n";
} else {
    die("Login failed.\n");
}

// Set currency conversion details
$date = '2023_04_05';
$baseCurrency = 'GBP';
$targetCurrency = 'EUR';
$amount = 1000;

// Set the date for which you want to retrieve the currencies
$day = '05';
$month = '04';
$year = '2023';

// Set the UUID of the currency you want to retrieve
$uuid = 'currency_uuid';

// Set the date for which you want to retrieve the daily average
$dailyAverageDate = '2023_04_10';

// Set the date range for which you want to retrieve the weekly average
$weeklyAverageFromDate = '2023_04_03';
$weeklyAverageToDate = '2023_04_07';

// Call the convert method to perform the currency conversion
$conversionResult = $fxdatapi->convert($username, $date, $baseCurrency, $targetCurrency, $amount);
if ($conversionResult) {
    echo "Conversion result: " . json_encode($conversionResult) . "\n";
} else {
    echo "Currency conversion failed: " . $conversionResult['error'] . "\n";
}

// Call the convertAll method to convert the base currency to all available target currencies
$convertAllResult = $fxdatapi->convertAll($username, $baseCurrency, $amount, $date);
if ($convertAllResult) {
    echo "Conversion to all currencies successful:\n";
    echo json_encode($convertAllResult) . "\n";
} else {
    echo "Conversion to all currencies failed: " . $convertAllResult['error'] . "\n";
}

// Call the getCurrencies method to get the available currencies for the given date
$currenciesResult = $fxdatapi->getCurrencies($username, $day, $month, $year);
if ($currenciesResult) {
    echo "Currencies for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($currenciesResult) . "\n";
} else {
    echo "Failed to retrieve currencies for the given date.\n";
}

// Call the getCurrency method to get the details of the specific currency for the given date
$currencyResult = $fxdatapi->getCurrency($uuid, $username, $day, $month, $year);
if ($currencyResult) {
    echo "Currency details for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($currencyResult) . "\n";
} else {
    echo "Failed to retrieve currency details for the given date.\n";
}

// Call the getHistoricalData method to get historical data for all currencies for the given date
$historicalDataResult = $fxdatapi->getHistoricalData($username, $date, $day, $month, $year);
if ($historicalDataResult) {
    echo "Historical data for " . $date . ":\n";
    echo json_encode($historicalDataResult) . "\n";
} else {
    echo "Failed to retrieve historical data for the given date.\n";
}

// Call the getHistoricalDataForCurrency method to get historical data for a specific currency
$historicalDataForCurrencyResult = $fxdatapi->getHistoricalDataForCurrency($uuid, $username, $day, $month, $year, $date);
if ($historicalDataForCurrencyResult) {
    echo "Historical data for currency " . $uuid . " on " . $date . ":\n";
    echo json_encode($historicalDataForCurrencyResult) . "\n";
} else {
    echo "Failed to retrieve historical data for the given currency and date.\n";
}

// Call the getDailyAverage method to get the daily average for the given date
$dailyAverageResult = $fxdatapi->getDailyAverage($dailyAverageDate);
if ($dailyAverageResult) {
    echo "Daily average for " . $dailyAverageDate . ":\n";
    echo json_encode($dailyAverageResult) . "\n";
} else {
    echo "Failed to retrieve daily average for the given date.\n";
}

// Call the getWeeklyAverage method to get the weekly average for the given date range
$weeklyAverageResult = $fxdatapi->getWeeklyAverage($weeklyAverageFromDate, $weeklyAverageToDate);
if ($weeklyAverageResult) {
    echo "Weekly average from " . $weeklyAverageFromDate . " to " . $weeklyAverageToDate . ":\n";
    echo json_encode($weeklyAverageResult) . "\n";
} else {
    echo "Failed to retrieve weekly average for the given date range.\n";
}

// Call the getMonthlyAverage method to get the monthly average for the given year and month
$monthlyAverageResult = $fxdatapi->getMonthlyAverage($monthlyAverageYear, $monthlyAverageMonth);
if ($monthlyAverageResult) {
    echo "Monthly average for " . $monthlyAverageYear . " and " . $monthlyAverageMonth . ":\n";
    echo json_encode($monthlyAverageResult) . "\n";
} else {
    echo "Failed to retrieve monthly average for the given year and month.\n";
}

// Call the getMarginsSpreads method to get the margins and spreads for the given date
$marginsSpreadsResult = $fxdatapi->getMarginsSpreads($username, $day, $month, $year);
if ($marginsSpreadsResult) {
    echo "Margins and spreads for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($marginsSpreadsResult) . "\n";
} else {
    echo "Failed to retrieve margins and spreads for the given date.\n";
}

// Call the getMarginSpread method to get the details of the specific margin or spread for the given date
$marginSpreadResult = $fxdatapi->getMarginSpread($uuid, $username, $day, $month, $year);
if ($marginSpreadResult) {
    echo "Margin or spread details for " . $day . "/" . $month . "/" . $year . ":\n";
    echo json_encode($marginSpreadResult) . "\n";
} else {
    echo "Failed to retrieve margin or spread details for the given date.\n";
}

// Call the getAllPerformances method to get all performances
$getAllPerformancesResult = $fxdatapi->getAllPerformances($username);
if ($getAllPerformancesResult) {
    echo "Performances list:\n";
    echo json_encode($getAllPerformancesResult) . "\n";
} else {
    echo "Failed to retrieve performances.\n";
}

// Set the UUID of the performance you want to retrieve
$performanceId = 'performance_id';

// Call the getPerformanceById method to get the details of a specific performance
$getPerformanceByIdResult = $fxdatapi->getPerformanceById($performanceId, $username);
if ($getPerformanceByIdResult) {
    echo "Performance details for " . $performanceId . ":\n";
    echo json_encode($getPerformanceByIdResult) . "\n";
} else {
    echo "Failed to retrieve performance details for the given ID.\n";
}

// Call the getAllSignals method to get all signals
$getAllSignalsResult = $fxdatapi->getAllSignals($username);
if ($getAllSignalsResult) {
    echo "Signals list:\n";
    echo json_encode($getAllSignalsResult) . "\n";
} else {
    echo "Failed to retrieve signals.\n";
}

// Set the UUID of the signal you want to retrieve
$signalId = 'signal_id';

// Call the getSignalById method to get the details of a specific signal
$getSignalByIdResult = $fxdatapi->getSignalById($signalId, $username);
if ($getSignalByIdResult) {
    echo "Signal details for " . $signalId . ":\n";
    echo json_encode($getSignalByIdResult) . "\n";
} else {
    echo "Failed to retrieve signal details for the given ID.\n";
}

?>
